zksync/src/20240805_AaveV3ZkSync_AaveV3ZkSyncActivation/AaveV3ZkSync_AaveV3ZkSyncActivation_20240805.t.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ProtocolV3TestBase} from 'aave-helpers/zksync/src/ProtocolV3TestBase.sol';
+import {AaveV3ZkSync_AaveV3ZkSyncActivation_20240805} from './AaveV3ZkSync_AaveV3ZkSyncActivation_20240805.sol';
+import {AaveV3ZkSync} from 'aave-address-book/AaveV3ZkSync.sol';
+import {MiscZkSync} from 'aave-address-book/MiscZkSync.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {IEmissionManager} from 'aave-v3-periphery/contracts/rewards/interfaces/IEmissionManager.sol';
+
+/**
+ * @dev Test for AaveV3ZkSync_AaveV3ZkSyncActivation_20240805
+ * command: FOUNDRY_PROFILE=zksync forge test --zksync --match-path=zksync/src/20240805_AaveV3ZkSync_AaveV3ZkSyncActivation/AaveV3ZkSync_AaveV3ZkSyncActivation_20240805.t.sol -vv
+ */
+contract AaveV3ZkSync_AaveV3ZkSyncActivation_20240805_Test is ProtocolV3TestBase {
+  AaveV3ZkSync_AaveV3ZkSyncActivation_20240805 internal proposal;
+
+  function setUp() public override {
+    vm.createSelectFork(vm.rpcUrl('zksync'), 41592552);
+    proposal = new AaveV3ZkSync_AaveV3ZkSyncActivation_20240805();
+
+    super.setUp();
+  }
+
+  /**
+   * @dev executes the generic test suite including e2e
+   */
+  function test_defaultProposalExecution() public {
+    defaultTest(
+      'AaveV3ZkSync_AaveV3ZkSyncActivation_20240805',
+      AaveV3ZkSync.POOL,
+      address(proposal)
+    );
+  }
+
+  function test_permissions() public {
+    assertFalse(AaveV3ZkSync.ACL_MANAGER.isPoolAdmin(MiscZkSync.PROTOCOL_GUARDIAN));
+    executePayload(vm, address(proposal));
+
+    assertTrue(AaveV3ZkSync.ACL_MANAGER.isPoolAdmin(MiscZkSync.PROTOCOL_GUARDIAN));
+    assertEq(
+      IEmissionManager(AaveV3ZkSync.EMISSION_MANAGER).getEmissionAdmin(proposal.ZK()),
+      proposal.ACI_MULTISIG()
+    );
+  }
+
+  function test_collectorHasUSDCFunds() public {
+    executePayload(vm, address(proposal));
```

Comment:
Why not doing this all together? Only executing payload once. It is more correct as all funds should be there